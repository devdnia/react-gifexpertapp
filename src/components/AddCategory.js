import React, { useState} from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ( { setCategories } ) => {
    
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) =>{
        setInputValue( e.target.value );

    };

    const handleSubmit = (e) =>{

        // Para prevenir el refresco por defecto del formulario
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            
            setCategories( cats => [inputValue, ...cats ] );
            setInputValue('');

        }

    };


    return (
        <form onSubmit={ handleSubmit }>
            <p>{ inputValue }</p>
           <input 
                type="text"
                value= { inputValue }
                onChange={ handleInputChange }
           />
        </form>
    )
}


AddCategory.prototype = {
    setCategories: PropTypes.func.isRequired
}
