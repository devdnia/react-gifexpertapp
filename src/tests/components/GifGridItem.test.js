import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


/**
 *  PRUEBAS:
 * 
 *  Prueba 1:
 *  1. Instalar Enzyme
 *  2. Instalar Enzyme to Json
 *  3. debe mostrar el componente correctamente
 *      * shallow
 *      * wrapper
 *      * wrapper .toMatchSnapshot()
 * 
 *  Prueba 2:
 *  1. Añadir PropTypes
 *     url, tittle = Obligatorios
 *  2. Enviar title y url a la hora de utilizar shallow()
 *  3. Actualizar su snapshot con los cambios
 */



describe('Prueba en GifGridItem', () => {

    const title = 'Un título';
    const url   = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={ title } url={ url } />);

    test('debe mostrar el componente correctamente (hacer el match con el snapshot )', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de tener un párrafo con el title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    });
    
    test('debe de tener la imagen igual a la url y alt de los props', () => {
       const img = wrapper.find( 'img');
       expect( img.prop( 'src' ) ).toBe( url ); 
       expect( img.prop( 'alt' ) ).toBe( title ); 
    });

    test('debe de tener animate__fadeIn', () => {
       const div = wrapper.find( 'div' );
       // console.log( div.prop( 'className') );
       const className = div.prop( 'className');
       expect( className.includes( 'animate__fadeIn') ).toBe( true );

    });
    
    
    
})
