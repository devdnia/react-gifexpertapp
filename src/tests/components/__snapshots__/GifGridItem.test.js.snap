// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en GifGridItem debe mostrar el componente correctamente (hacer el match con el snapshot ) 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
